refs #279 Improve coverage

e1Ru1o opened the pull request:
refs #279 

**Changes:**
-  Add more tests mainly related with files not covered by the actual testsuit
- Aggregate test coverage data correctly so as to include multiple packages in HTML reports

**Does this change need to mentioned in CHANGELOG.md?**
no

**Requires testing**
no

e1Ru1o committed:
[makefile] Change linux and windows BINPATH value

e1Ru1o committed:
[test] refs #279 - Add test for SkycoinPexNode

e1Ru1o committed:
[test] refs #279 - Add test for SkycoinPexNodeIterator

e1Ru1o committed:
[mocks] Update Address mock

e1Ru1o committed:
[test] refs #279 - Add initial test for NewGenericOutput

Pending:
-  Find how to register skycoin coins without import the skycoin package, import it produces cyclic imports

e1Ru1o committed:
[test] refs #279 - Add test for ParseDate

e1Ru1o committed:
[test] refs #279 - Add test for EmptyPassword and ConstantPassword

Note:
-  TestPassword cover both funtions

e1Ru1o committed:
[util] - Update FormatUint64

Remove negativity verification inside the function. uint64 can not be negative

e1Ru1o committed:
[test] refs #279 - Add initial test-suit for util.go file 

Pendig:
-   `GetCoinValue`
-   `AddressFromString`
-    Both need some previously registered coins

e1Ru1o committed:
[test] refs #279 - Add test-suit for wallet.go and update test for NewGenericOutput

e1Ru1o committed:
[test] refs #279 - Improve tests related with account.go

e1Ru1o committed:
[test] refs #279 - Improve tests related to blockchain.go

stdevMac committed:
Merge remote-tracking branch 'origin/develop' into stdevRulo_t279_improve_coverage

e1Ru1o committed:
[coin] - Fix some errors, and update account_test.go

Changes:
-   `SkycoinUninjectedTransaction.ComputeFee(...)` method report invalid ticker error when the ticker in not supported
- 
  `SkycoinCreatedTransactionInput.GetSpentOutput()` was calling `GetCoin(...)` with an invalid argument, know use the correct one
-   `TestWalletsListAssets` modified

e1Ru1o committed:
Merge remote-tracking branch 'simelo/stdevRulo_t279_improve_coverage' into stdevRulo_t279_improve_coverage

e1Ru1o committed:
[network] - merge networking.go into network.go

e1Ru1o committed:
Merge remote-tracking branch 'origin/develop' into stdevRulo_t279_improve_coverage

e1Ru1o committed:
[test] refs #279 - Fix a lint error

olemis reviewed (changes requested):
needs more simple tests added

olemis committed:
[mocks] refs #279 - Update coin mocks

  modified:   src/coin/mocks/Address.go
  modified:   src/coin/mocks/AltcoinPlugin.go
  new file:   src/coin/mocks/PubKey.go
  new file:   src/coin/mocks/SecKey.go
  modified:   src/coin/mocks/WalletSet.go

olemis committed:
[util] refs #279 - Tests for adding fake altcoin plugin

olemis committed:
[util] refs #279 - Test util.AltcoinQuotient

olemis committed:
[util] refs #279 - Tests for unregistered altcoin plugin

olemis committed:
[ci] refs #279 - Aggregate multi-step coverage consistently

olemis reviewed (changes requested):
The following functions are untested 

- `util:GetCoinValue`
- `util:AddressFromString`
- `util:KeyValuesWithDefaults.SetValue`
- `util:LookupSignServiceForWallet`
- `util:GenericMultiWalletSign`
- `util:SignTransaction`
- `util:ReadyForTxn`
- `util:LookupSignerByUID` ... I'd rather move this one to `sign.go` <= @stdevMauricio1802 what d'u think ?
- `util:GenericOutput.PushCoins`
- `skycoin:WalletDirectory.GetStorage`
- `skycoin:SkycoinLocalWallet.ListWallets`
- `skycoin:SkycoinLocalWallet.Encrypt`
- `skycoin:SkycoinLocalWallet.Decrypt`
- `skycoin:SkycoinLocalWallet.IsEncrypted`
- `skycoin:LocalWallet.ReadyForTxn`
- `skycoin:LocalWallet.GetSignerUID`
- `skycoin:LocalWallet.GetSignerDescription`
- `skycoin:SkyFiberPlugin:ListSupportedFamilies`
- `skycoin:SkyFiberPlugin.GetName`
- `skycoin:SkyFiberPlugin.GetDescription`
- `skycoin:SkyFiberPlugin` entry points
- `skycoin:SkyFiberPlugin.PubKeyFromBytes` , please add a helper in `util` for this ... and ensure it's covered by tests
- `skycoin:SkyFiberPlugin.SecKeyFromBytes` , please add a helper in `util` for this ... and ensure it's covered by tests

Internal paths (e.g. branches for error conditions) are not tested in the following functions 

- `util:FormatCoins`
- `util:NewKeyValuesWithDefaults`
- most functions implementing wallet contract for Skycoin altcoin
  * ... notably `skycoin:SkycoinLocalWallet.GetWallet`
-

olemis committed:
[vendor] refs #279 - Update go deps

olemis committed:
[sky] refs #279 - Implement TestSupportedAssets

... import test helper functions from stretchr/testify

- ObjectsAreEqual
- Fail
- FasilNow
- ElementsMatch
- IsEmpty
- MessageFromMsgAndArgs
- LabeledContent
- IndentMessageLines
- LabeledOutput

olemis committed:
[util] refs #279 - Fix lint errors

... reported in https://travis-ci.org/fibercrypto/fibercryptowallet/jobs/633677596#L445-L456

olemis committed:
[util] refs #279 - Fix bytes:Buffer.WriteString returns two values

olemis committed:
[sky] refs #279 - TestPendingTxnStatus

olemis committed:
[sky] refs #279 - Test there is no fee in Sky

olemis committed:
[sky] refs #279 - Test that uninjected transactions do not support CalculatedHour ticker

olemis committed:
[sky] refs #279 - Request fee for invalid asset from pending txn should return error

olemis committed:
[sky] refs #279 - Add TestUninjectedTxnTimestamp

olemis committed:
[sky] refs #279 - Add TestUninjectedTxnStatus

olemis committed:
[sky] refs #279 - Add TestUninjectedTxnFee

olemis committed:
[sky] refs #279 - Fix linter error

... reported in https://travis-ci.org/fibercrypto/fibercryptowallet/jobs/634419320#L446-L456

olemis committed:
[sky] refs #279 - May not request transaction fee in CalculatedHour

olemis committed:
[sky] refs #279 - Assert error when requesting fee in CalculatedHour

... for uninjected and pending transaction

olemis committed:
[sky] refs #279 - Skycoin transaction instances do not support CalculatedHour

olemis committed:
[sky] refs #279 - Assertions for assets supported by SkycoinTransaction SkycoinCreatedTransaction

olemis committed:
[sky] refs #279 - Add TestSkycoinTxnFee

olemis committed:
[sky] refs #279 - Add TestSkycoinCreatedTxnFee

olemis committed:
[sky] refs #279 - Assert assets supported by instances representing transaction inputs and outputs

olemis committed:
[sky] refs #279 - Assert SkycoinCreatedTransactionInput supports all assets defined by SkyFiber altcoin plugin

e1Ru1o committed:
[test] refs #279 - Add TestLookupSignerByUID

e1Ru1o committed:
[test] refs #279 - Fix one error in the previous commit

Remove duplicated imports

e1Ru1o committed:
[util][test] refs #279 - Add TestGetCoinValue

Fix:
- Deleted duplicated method `AttachSigner`
- Return the correct error in `GetCoinValue` when is not possible to parse `coin`

e1Ru1o committed:
[test] refs #279 - Improve TestFormatCoins

e1Ru1o committed:
[test] refs #279 - Add TestAddressFromString 

Delete an unused variable

e1Ru1o committed:
[test] refs #279 - Improve TestNewGenericOutput

e1Ru1o committed:
[test] refs #279 - Fix an error

variable unused because a naming error

e1Ru1o committed:
[test] refs #279 - Fix name error

Fix word Strings

e1Ru1o committed:
[test] refs #279 - Remove a signer in TestLookupSignerByUID which caused a failure in TestSignersReadyForTxn

e1Ru1o committed:
[test] refs #279 - Change a method call in TestAddressFromString which cause an error

e1Ru1o committed:
[test] refs #279 - Fix wrong parameters passed to mock in TestAddressFromString

e1Ru1o committed:
[test] refs #279 - Improve TestSignersReadyForTxn

Details:
-  Previosly the test only covered SignServicesForTxn
-  Now also covers ReadyForTxn

e1Ru1o committed:
[test] refs #279 - Add TestLookupSignServiceForWallet

e1Ru1o committed:
[test] refs #279 - Fix failures on Macos


`pulgin_test.go`:
-   merge variable declaration with assignment on next line

`coin_test.go`:
-  change some names for removing underscores
-  check return value of `output.PushCoins`

`sig_test.go`:
-  Add nolint comment in `RemoveSingService`

e1Ru1o committed:
[test] refs #279 - Add missing import in `sign_test.go`

e1Ru1o committed:
[test] refs #279 - Fix Macos errors

Errors:
-  `src/util/plugin_test.go:61:25`: Error return value of `RemoveSignService` is not checked
-   `src/util/sign_test.go:168:2`: S1021: should merge variable declaration with assignment on next line

e1Ru1o committed:
[test] refs #279 - Add `TestSignTransaction`

e1Ru1o committed:
[test] refs #279 - Added missing import in `sign_test.go`

e1Ru1o committed:
[test] refs #279 - Fix an error on `TestSignTransaction`

not enough arguments in call to `pwdReader`

e1Ru1o committed:
[test] refs #279 - Pass the correct parameters to `pwdReader` in `TestSignTransaction`

e1Ru1o committed:
[test] refs #279 - Fix error whe use `mock.Anything` in `TestSignTransaction`

e1Ru1o committed:
[test] refs #279 - Add TestGenericMultiWalletSign

e1Ru1o committed:
[test] refs #279 - Add `TestKeyValuesWithDefaultsSetValue`

e1Ru1o committed:
[test] refs #279 - Add `TestMessageFromMsgAndArgs`

e1Ru1o committed:
[test] refs #279 - Add `TestIndentMessageLines`

e1Ru1o committed:
[test] refs #279 - Add `TestLabeledOutput`

e1Ru1o committed:
[test] refs #279 - Fix lint error

proccess is a misspelling of process

e1Ru1o committed:
[test] refs #279 - Add `TestWalletDirectoryGetStorage`

e1Ru1o committed:
[test] refs #279 - Add `TestSkycoinLocalWalletListWallets`

e1Ru1o committed:
[test] refs #279 -  Add `TestSkycoinLocalWalletEncrypt`

e1Ru1o committed:
[wallet] - Fix an error in `checkEquivalentSkycoinWallets`

Description:
-    Method `Value` of an iterator always raise an error if not exists a previous call to the `Next` method

e1Ru1o committed:
[test] refs #279 - Add `TestSkycoinLocalIsEncrypted`

e1Ru1o committed:
[test] refs #279 - Add `TestSkycoinLocalWalletDecrypt`

e1Ru1o committed:
[test] refs #279 -  Add `TestLocalWalletSetLabel`

e1Ru1o committed:
[test] refs #279 - Add `TestWalletsReadyForTxn`

e1Ru1o committed:
[test] refs #279 - Add `TestLocalWalletFunctions`

e1Ru1o committed:
[test] refs #279 - Add `TestWalletNodeGetWalletSet` and `TestWalletNodeGetStorage`

e1Ru1o committed:
[test] refs #279 - Add tests related to `SeedService`

-  `TestSeedServiceGenerateMnemonic`
-  `TestSeedServiceVerifyMnemonic`

e1Ru1o committed:
[test] refs #279 - Add `TestErrorTickerInvalidError`

e1Ru1o committed:
[test] refs #279 - Change `TestLocalWalletFunctions` to `TestWalletFunctions`

Changes:
-   Test `RemoteWallet` functions too

e1Ru1o committed:
[test] refs #279 - Fix lint errors

e1Ru1o committed:
[test] refs #279 - Add `TestSkyFiberPluginRegisterTo` and `TestSkyFiberPluginFunctions`

e1Ru1o committed:
[test] refs #279 - Add `TestSkyFiberPluginNetOperations`

e1Ru1o committed:
[test] refs #279 - Update `TestSkyFiberPluginFunctions`

Changes:
-   Include `Params` verification
-   Include `LoadSignService` method test

e1Ru1o committed:
[test] refs #279 - Add `TestSkyFiberPluginAddressFromString`

e1Ru1o committed:
[test] refs #279 - Add and test util helpers for `PubKeyFromBytes` and `SecKeyFromBytes`

e1Ru1o committed:
[test] refs #279 - Add `TestSkyFiberPluginPubKeyFromBytes` and `TestSkyFiberPluginSecKeyFromBytes`

e1Ru1o committed:
[test] refs #279 - Fix lint error

ineffectual assignment

e1Ru1o committed:
[test] refs #279 - Improve coverage in `account.go`

more internal paths covered by tests

e1Ru1o committed:
[test] refs #279 - Deleted old `SkycoinApiMock` implementation and use `mockery` to generate it instead

e1Ru1o committed:
[blockchain] - Removed an unnecessary conditional

e1Ru1o committed:
[test] refs #279 - Improve coverage in `blockchain.go`

All testable internal paths covered

e1Ru1o committed:
[test] refs #279 - Improve coverage in `cipher.go`

100% covered

e1Ru1o committed:
[test] refs #279 - Solve lint errors

olemis committed:
Merge remote-tracking branch 'fc/develop' into stdevRulo_t279_improve_coverage

olemis committed:
[ci] refs #279 - Run make test-data against temp coverage profile and merge with coverage output

olemis committed:
[vendor] refs #279 - Add bcrypt under vendor/

olemis committed:
[addressbook] refs #279 - Address book imports official Skycoin repository

olemis committed:
[addressbook] refs #279 - Fix import skycoin/SkycoinProject => SkycoinProject/skycoin

olemis committed:
[sky] refs #279 - Fix linter errors

... reported in https://travis-ci.org/fibercrypto/fibercryptowallet/jobs/640667694#L453-L466

olemis committed:
[sky] refs #279 - Fix linter errors

... reported in https://travis-ci.org/fibercrypto/fibercryptowallet/jobs/640700253#L448-L455
